src/libraries/System.Net.Quic/tests/FunctionalTests/QuicConnectionTests.cs

```diff
@@ -31,15 +35,27 @@ public async Task TestConnect()
             await using QuicConnection serverConnection = acceptTask.Result;
             await using QuicConnection clientConnection = connectTask.Result;
 
-            Assert.Equal(listener.LocalEndPoint, serverConnection.LocalEndPoint);
-            Assert.Equal(listener.LocalEndPoint, clientConnection.RemoteEndPoint);
-            Assert.Equal(clientConnection.LocalEndPoint, serverConnection.RemoteEndPoint);
+            IgnoreScopeIdIPEndpointComparer endPointComparer = new();
+            Assert.Equal(listener.LocalEndPoint, serverConnection.LocalEndPoint, endPointComparer);
+            Assert.Equal(listener.LocalEndPoint, clientConnection.RemoteEndPoint, endPointComparer);
+            Assert.Equal(clientConnection.LocalEndPoint, serverConnection.RemoteEndPoint, endPointComparer);
```

Comment:
To improve confidence in my changes, I extended this test to test IPv6. `serverConnection.LocalEndPoint` has a ScopeId for some reason while `listener.LocalEndpoint` doesn't. I assume it is asigned after binding, and we can ignore this difference.

Reply:
scopeID should be only relevant for LLA. I'm wondering if we are deviating from Socket/NetworkStream here because the underlying MSQuic implementation.

Reply:
It's possible. If we think it's a problem we can track it in a separate issue.

Reply:
It seems like this was existing condition. We miss it because QuicConnectionTests.TestConnect only tests IPv4. I sync back and I tried it with IPv6 and it also fails:
```
   Starting:    System.Net.Quic.Functional.Tests (parallel test collections = on, max threads = 4)
      System.Net.Quic.Tests.QuicConnectionTests.TestConnect [STARTING]
      System.Net.Quic.Tests.QuicConnectionTests.TestConnect [FAIL]
        Assert.Equal() Failure
        Expected: [::1]:34850
        Actual:   [::1%1]:34850
```

I will fix it tomorrow. 
(cc: @manickap)